src/bam/read_orientation.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021 Johannes Köster.
+// Licensed under the MIT license (http://opensource.org/licenses/MIT)
+// This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use strum_macros::AsRefStr;
+
+use crate::bam;
+
+/// Representation of read orientation
+/// (e.g. F1R2 means that the forward read comes first on the reference contig,
+/// followed by the reverse read, on the same contig).
+#[derive(Debug, Clone, Copy, PartialEq, Eq, AsRefStr)]
+#[cfg_attr(feature = "serde_feature", derive(Serialize, Deserialize))]
+pub(crate) enum ReadOrientation {
+    F1R2,
+    F2R1,
+    R1F2,
+    R2F1,
+    F1F2,
+    R1R2,
+    F2F1,
+    R2R1,
+    None,
+}
+
+impl<'a> From<&'a bam::Record> for ReadOrientation {
```

Comment:
If the only way to get this orientation is using the from method, why is it in rust-htslib?

Reply:
I see read orientation as a property of the mapped record. Hence, although it has to be inferred from the bam, it made sense to me to have it in here. We could have it in a separate crate, but that would need to be an entirely new one. It cannot be in rust-bio (because then rust-bio would need to depend on rust-htslib). It also cannot be in rust-bio-types, because rust-htslib depends on rust-bio-types. I think it is a reasonable convenience functionality that can be safely added here without bloating the codebase too much.

Reply:
Sure, I get that, but then I would expect  a method on bam::Record itself. Also using the result of this function will probably involve another match statement which might not be the most optimal approach.

Reply:
Thinking about it, we could move `ReadOrientation` to bio-types, and implement instead `Into<ReadOrientation>` here in rust-htslib.

Reply:
That sounds reasonable. Should it be called ReadPairOrientation? The read orientation is represented by the flag is_reverse.

Reply:
Very good idea as well. I'll do that.